map-idling-resource/library/src/main/java/com/novoda/espresso/MapIdlingResource.java

```diff
@@ -0,0 +1,78 @@
+package com.novoda.espresso;
+
+import android.support.annotation.Nullable;
+import android.support.test.espresso.IdlingResource;
+import android.support.test.espresso.core.deps.guava.base.Preconditions;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentManager;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+
+import java.util.List;
+
+/**
+ * An IdlingResource that finds the {@link SupportMapFragment} in given {@link FragmentActivity}
+ * and hook itself to map's async ready callback.
+ */
+public class MapIdlingResource implements IdlingResource, OnMapReadyCallback {
+
+    private ResourceCallback callback;
+    private boolean isMapReady = false;
+
+    public static IdlingResource from(final FragmentActivity activity) {
+        FragmentManager fragmentManager = activity.getSupportFragmentManager();
+        SupportMapFragment mapFragment = findSupportMapFragment(fragmentManager);
+
+        Preconditions.checkNotNull(mapFragment, "Couldn't find SupportMapFragment in " + activity);
+
+        return MainThread.createOnMainThread(mapFragment);
+    }
+
+    @Nullable
+    private static SupportMapFragment findSupportMapFragment(FragmentManager supportFragmentManager) {
+        List<Fragment> fragments = supportFragmentManager.getFragments();
+        for (Fragment fragment : fragments) {
+            if (fragment instanceof SupportMapFragment) {
+                return (SupportMapFragment) fragment;
```

Comment:
Yep, this assumes that you wait for only 1 map initialization. 
There are apps that have multiple maps in 1 screen. But in this case, they are mostly for display purposes. 
Either I should make that assumption very clear. Or I can hook into all MapFragment's available in the `FragmentManager`
What do you think?